src/app/Shared/Services/Api.service.tsx

```diff
@@ -474,13 +480,22 @@ export class ApiService {
           method: 'POST',
           body,
           headers,
-          selector: response => response.text(),
           signal,
         });
       }),
+      concatMap(v => {
+        if (v.ok) {
+          return from(v.text());
+        }
+        throw from(v.text());
+      }),
       tap({
         next: () => window.onbeforeunload = null,
-        error: () => window.onbeforeunload = null,
+        error: (err) => {
+          window.onbeforeunload = null;
+          err.subscribe(msg => this.notifications.danger('Upload Failed', msg, NotificationCategory.ArchivedRecordingCreated));
```

Comment:
Is the Re-Upload Recording modal intended to stay open if you upload an invalid file? If so, I think the error toast gets hidden while the modal is open.

Reply:
Hmm. The toast is still visible for me while the modal is open...

![image](https://user-images.githubusercontent.com/3787464/159574366-f8953f84-bd25-48ad-a5e0-eaa65dafe4b7.png)

Reply:
It's still visible for me as well.

Reply:
Ah - I didn't realize that the "Enable or disable notifications by category" feature preserves your settings even after I restarted the backend. I disabled a bunch of notifications while testing another feature a few days ago.

If the `ArchivedRecordingUploaded` notification is disabled, I don't see the error message.

Reply:
Oh, yea. The notification settings go into browser local storage.

Reply:
I think this is good enough as is - I can still see the red notification badge when I close the modal.